### ISSUE TITLE
README.md files need a better way to store header/footer info

### ISSUE THREAD

**Issue by tjayrush** [Great-Hill-Corporation/quickBlocks#89]
Every tool has a file called options.cpp where the command line options are parsed.

Currently, the header and footer of each tool's README.md files is hard coded into the options.cpp files. Here's an example: https://github.com/Great-Hill-Corporation/quickBlocks/blob/master/src/tools/ethName/options.cpp#L93

This is a bad poor idea as I must edit the source, and re-build the application to change the README.md file.

The reason I did it this way is because the 'Usage' part of the README.md file is generated based on the actual options defined in the code. The Usage is presented when the user enters an incorrect command line option and does not include the header and footer.

I need a way to both autogenerate the Usage part of the README.md and be able to edit the header and footer parts of the README file from a text file or something.

**Comment by crodnun** [comment]
Hi Thomas , what do you think about this idea:

1) Create a new folder under docs/templates  (templates would be a new folder at current structure)
2) Put there the same structure of directories you have at src. e.g. tools apps. .... etc every piece that can have a README.md file will have a template (TEMPLATE.txt or whatever you want) under this directory.

   An example for a single tool:
   
    docs/templates
    docs/templates/tools/ethName
    docs/templates/tools/ethName/template.txt

3) The format of the template txt file would be the following, with 3 sections:

     3-1) The header text, moved from the options.cpp file here:

     Text with the header ..... blah blah....
     More Header text .....
   
     3-2) An univocal tag, to be replaced by the toold/command help text
    
      3-3) The footer text, again the same you have at options.cpp moved here and no longer present at codes.

4) Then when running: make readme or any other target you can think on, we re-generate the latest README.md (some scripting here in the middle, with python) using the template and replacing the TAG with the latest command usage.

Find attached an example of how it could look like, using the ethName example and any TAG defined by us.

[template.txt](https://github.com/Great-Hill-Corporation/quickBlocks/files/1204793/template.txt)

**Comment by tjayrush** [comment]
Very good idea. I would rather store the documentation template in a folder at each tool. For example, ./src/tools/ethName/docs/README.tmpl. (I've used .tmpl files elsewhere.)

In this way, if I'm working on the ethName tool, I don't have to move to a different folder to document it. Plus--in this way, we can preserve the ./quickBlocks/docs folder for 'output' only documentation (i.e. finished docs). In a sense, I'm considering the README.tmpl file source code.

For the time being use this syntax for the 'univolcal tag.'  [{TAG}]     This would agree with all other TAGGING I've done throughout the project.

You may go ahead and make whatever changes you need to make for the ethName project. My repo is pushed and up to date.

**Comment by tjayrush** [comment]
Very nice. I made a few small changes. 

1. I renamed README.tmpl to README.tmpl.md is the ./ethName/docs folder so I could more easily use a markdown editor to edit this file,
2. I change the comment to -th instead of -h since the tools' help command generates markdown with -th.
3. I made it so the 'footer' gets generated programatically. This is because all tools have the same default options, and they sometimes change. In this way, I don't have to edit every template if a new default option is added.
4. I removed the COptions::header and COptions::footer from the code and fixed the three or four existing tools that used this "old" feature.
5. 

From now on, we will use this "new" template feature for making READMEs.

I am going to close this bug since the feature is now complete, but we still have to fully document all the tools/apps. I won't create a new issue for that.
